enhancements/catalog-data-transfer-reduction.md

```diff
@@ -128,295 +120,134 @@ Some terminology:
+This example extends to package-versions by replacing GVK with
+some version range and sorting by bundle version to find the latest versions
+that satsify all ranges.
 
 #### Merging catalogs
 
-Now that the diff `output` has been created, some mirroring command like `oc adm catalog mirror`
-will send the `output` (DC) and all referenced images to a disk mount.
-The mount is then transferred via meatspace to the disconnected side,
-and images are again mirrored from the mount to a registry.
+Now that the diff (catalog `B`) has been created, some command
+like `oc adm catalog mirror` can write the DC and all referenced images to a disk mount.
+The mount's data is then transferred by hand to the disconnected side,
+and images are pushed to a catalog registry, making the partial catalog image available.
 
-If there is no existing catalog in the cluster (fresh cluster),
+If there is no existing catalog in the cluster (fresh cluster, mode `headsOnly`),
 then a CatalogSource and set of Subscriptions are created and the process is complete.
 
-If one or more catalogs do exist in the cluster, then a new CatalogSource
-must be created for `output`. This is already possible to do.
-Since multiple catalogs can serve the same bundle, duplicated dependencies
-do not throw errors. All replaces will point to bundles in the cluster's catalog state
-by the algorithm.
-
-#### Automation
-
-Since this workflow is disjointed by the meatspace data transfer, it cannot be fully automated.
-However, the workflow stages on the connected and disconnected sides can be automated separately.
-
-On the connected side, the `opm diff` command could sit behind a server,
-receiving requests for diffs. On receiving a request, the server pulls `input_new`
-from its image registry and runs `opm diff` with the request specifying `input_old`,
-optionally `input_expand`, and the desired modes and options.
-The response is the generated diff, and any extra data required for further diff action
-(see [open question 1a](#open-questions-optional)).
-
-On the disconnected side:
-- `input_old` can be created periodically (cronjob) from the cluster’s state
-by the catalog snapshot tool and placed in some mounted directory, priming it for meatspace transfer.
-- `output` and some catalog data (potentially an entire CatalogSource)
-can be placed in some directory being watched by a daemon
-that will perform any necessary catalog add/merge behavior,
-then create/update a CatalogSource in-cluster.
-    - This may have security implications.
+If one or more catalogs do exist in the cluster (existing cluster, mode `latest`),
+then there are two options for merging `A` and `B`:
+1. A new CatalogSource is be created for `B` to live alongside that of `A`.
+OLM's resolver will [prioritize][cs-priority] bundles from `B` over counterparts in `A`.
+All replaces will point to bundles in either `A` or `B` by the algorithm.
+2. Pull the existing index image for `A` (retrieved from a CatalogSource field)
```

Comment:
This is the option we need to use. First, because existing customers will have problems since `Subscriptions` usually point to a specific catalog by name. Second, because otherwise customers end up with dozens of catalogs and it's unclear what relationship they have with another.